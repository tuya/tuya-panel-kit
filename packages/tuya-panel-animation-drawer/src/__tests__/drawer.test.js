/**
 * @jest-environment jsdom
 */
import React from 'react';
import { View, Image, Animated } from 'react-native';
import { shallow } from 'enzyme';
import Drawer from '../index';

const winWidth = 375;
const winHeight = 667;
const imgUrlList =
  'https://images.tuyacn.com/rms-static/3f5ccbb0-94bc-11ea-8e3b-e3ecba4013ec-1589334682603.png?tyName=speed1.png';
const renderContent = (
  <View
    style={{
      width: '100%',
      height: '100%',
      backgroundColor: 'red',
    }}
  >
    <View
      style={{ flexDirection: 'row', justifyContent: 'center', alignItems: 'center' }}
      key={imgUrlList}
    >
      <Image source={{ uri: imgUrlList }} style={{ width: 100, height: 100 }} />
    </View>
  </View>
);

jest.mock('Animated', () => {                                                                                                                                                                         
  const ActualAnimated = require.requireActual('Animated');                                                                                                                                           
  return {                                                                                                                                                                                            
    ...ActualAnimated,                                                                                                                                                                                
    parallel: (value, config) => {                                                                                                                                                                      
      return {                                                                                                                                                                                        
        start: jest.fn(),                                                                                                                                                  
      };                                                                                                                                                                                              
    },                                                                                                                                                                                                
  };                                                                                                                                                                                                  
}); 

describe('Drawer components', () => {
  it('basic render', () => {
    let placement = 'left';
    let visible = true;
    const wrapper = shallow(
      <Drawer
        width={winWidth / 3}
        height={winHeight}
        placement={'left'}
        visible={visible}
        onMaskPress={jest.fn()}
        onStateChange={jest.fn()}
        renderContent={renderContent}
      />
    );
    const pander = wrapper.findWhere(
      c => c.name() === 'AnimatedComponent' && !!c.prop('onStartShouldSetResponder') === true
    );
    const instance = wrapper.instance();
    instance.range = 375;
    pander.simulate('startShouldSetResponder', { nativeEvent: { locationX: 45, locationY: 3 } });
    pander.simulate('moveShouldSetResponder');
    pander.simulate('responderGrant', {
      nativeEvent: { touches: [] },
      touchHistory: { touchBank: [] },
    });
    pander.simulate('responderMove', {
      nativeEvent: { touches: [] },
      touchHistory: { touchBank: [] },
    });
    pander.simulate('responderRelease', {
      nativeEvent: { touches: [] },
      touchHistory: { touchBank: [] },
    });
    expect(wrapper).toMatchSnapshot();
    placement = 'right';
    visible = false;
    wrapper.setProps({ placement, visible });
  });
});
