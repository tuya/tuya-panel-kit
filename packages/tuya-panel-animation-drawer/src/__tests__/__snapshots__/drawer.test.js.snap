// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer components basic render 1`] = `
<View
  style={
    Array [
      Object {
        "left": 0,
        "position": "absolute",
        "top": 0,
      },
      Object {},
    ]
  }
>
  <AnimatedComponent
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "top": 0,
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0.6)",
          "height": 1334,
          "position": "absolute",
          "width": 750,
        },
        Object {
          "left": 0,
          "opacity": 1,
        },
      ]
    }
  />
  <AnimatedComponent
    style={
      Object {
        "flexDirection": "row",
        "left": 0,
        "position": "absolute",
        "top": 0,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F8F8F8",
          },
          Object {
            "height": 667,
            "width": 125,
          },
        ]
      }
    />
  </AnimatedComponent>
</View>
`;
